subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskExecutionAccessChecker.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks;
+
+import org.gradle.api.internal.TaskInternal;
+
+public interface TaskExecutionAccessChecker {
```

Comment:
This interface probably deserves some javadoc

Reply:
Fixed in https://github.com/gradle/gradle/pull/23210/commits/7a7d164ec05ff0df0e79a79334f9fdfb322f65f5